src/common/vector_operations/vector_cast.cpp

```diff
@@ -168,10 +168,126 @@ static bool NumericCastSwitch(Vector &source, Vector &result, idx_t count, strin
 	}
 }
 
+template <class T>
+bool FillEnum(string_t *source_data, ValidityMask &source_mask, const LogicalType &source_type, T *result_data,
+              ValidityMask &result_mask, const LogicalType &result_type, idx_t count, string *error_message) {
+	bool all_converted = true;
+	if (source_mask.AllValid()) {
+		for (idx_t i = 0; i < count; i++) {
+			auto string_value = source_data[i].GetString();
+			auto pos = EnumType::GetPos(result_type, string_value);
+			if (pos == -1) {
+				if (!error_message) {
+					throw CastException(source_type, result_type);
+				} else {
+					all_converted = false;
+					result_mask.SetInvalid(i);
+				}
+			} else {
+				result_data[i] = pos;
+			}
+		}
+	} else {
+		for (idx_t i = 0; i < count; i++) {
+			if (source_mask.RowIsValid(i)) {
+				auto string_value = source_data[i].GetString();
+				auto pos = EnumType::GetPos(result_type, string_value);
+				if (pos == -1) {
+					if (!error_message) {
+						throw CastException(source_type, result_type);
+					} else {
+						all_converted = false;
+						result_mask.SetInvalid(i);
+					}
+				} else {
+					result_data[i] = pos;
+				}
+			} else {
+				result_mask.SetInvalid(i);
+			}
+		}
+	}
+	return all_converted;
+}
+
+template <class T>
+bool TransformEnum(Vector &source, Vector &result, idx_t count, string *error_message) {
+	D_ASSERT(source.GetType().id() == LogicalTypeId::VARCHAR);
+	auto enum_name = EnumType::GetTypeName(result.GetType());
+	bool all_converted = true;
+	switch (source.GetVectorType()) {
+	case VectorType::CONSTANT_VECTOR: {
+		result.SetVectorType(VectorType::CONSTANT_VECTOR);
+		auto source_data = ConstantVector::GetData<string_t>(source);
+		auto result_data = ConstantVector::GetData<T>(result);
+		if (ConstantVector::IsNull(source)) {
+			ConstantVector::SetNull(result, true);
+		} else {
+			ConstantVector::SetNull(result, false);
+			auto string_value = source_data[0].GetString();
+			auto pos = EnumType::GetPos(result.GetType(), source_data[0].GetString());
```

Comment:
This code path seems repeated in several places, can we extract it into a separate function?